src/targets/maven.ts

```diff
@@ -0,0 +1,229 @@
+import { TargetConfig } from '../schemas/project_config';
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { BaseTarget } from './base';
+import { ConfigurationError } from '../utils/errors';
+import { homedir } from 'os';
+import * as path from 'path';
+import * as fs from 'fs';
+import { exec } from 'child_process';
+import { isDryRun } from '../utils/helpers';
+
+// TODO: add docs to the readme
+
+const GRADLE_PROPERTIES_FILENAME = 'gradle.properties';
+
+const ANDROID_DIST_EXTENSION = '.aar'; // Must include the leading `.`
+const ANDROID_RELEASE_SUBSTR = 'release';
+
+/** Config options for the "maven" target. */
+interface MavenTargetConfig {
+  // Required env variables
+  ossrhUsername: string;
+  ossrhPassword: string;
+  mavenUsername: string;
+  mavenPassword: string;
+  // Optional env variables (have a default value)
+  distributionsPath: string;
+  settingsPath: string;
+  mavenRepoUrl: string;
+  mavenRepoId: string;
+  mavenCliPath: string;
+  gradleCliPath: string;
+}
```

Comment:
```suggestion
/** Config options for the "maven" target. */
type MavenTargetConfig = {
    [key in keyof ((typeof RequiredConfig) & (typeof OptionalConfig))]: string
}
```

Reply:
This seems easier than merging enum types 😁.